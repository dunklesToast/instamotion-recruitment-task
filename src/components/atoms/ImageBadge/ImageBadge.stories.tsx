import React from 'react';
import type { Meta, Story } from '@storybook/react';
import type { ImageBadgeProps } from './ImageBadge';
import { ImageBadge } from './ImageBadge';

export default {
  title: 'Atoms/ImageBadge',
  component: ImageBadge,
} as Meta;

const ImageBadgeComponent: Story<ImageBadgeProps> = (args) => <ImageBadge {...args} />;

export const Default = ImageBadgeComponent.bind({});
Default.args = {
  children: 'Story of ImageBadge',
};
