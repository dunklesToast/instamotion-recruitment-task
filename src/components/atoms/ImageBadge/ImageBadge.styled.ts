import { colors } from '@style/colors';
import styled from 'styled-components';

export const ImageBadge = styled.div`
  background-color: ${colors.primary};
  border-bottom-left-radius: 4px;
  border-top-left-radius: 4px;
  color: white;
  display: inline-block;
  padding: 8px 32px 8px 8px;
`;
