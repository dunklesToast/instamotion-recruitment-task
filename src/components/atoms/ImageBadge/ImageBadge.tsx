import React from 'react';
import { Text } from '@atoms/Text/Text';
import * as S from './ImageBadge.styled';

export interface ImageBadgeProps {
  /** The Image Count. */
  children: React.ReactNode;
  className?: string;
}

/**
 * Description of ImageBadge.
 */
export function ImageBadge({ children, className }: ImageBadgeProps): JSX.Element {
  return (
    <S.ImageBadge className={className} data-testid='imagebadge'>
      <Text>{children}</Text>
    </S.ImageBadge>
  );
}
