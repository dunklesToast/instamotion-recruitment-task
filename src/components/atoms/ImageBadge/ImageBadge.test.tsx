import React from 'react';
import { render, screen } from '@testing-library/react';
import { ImageBadge } from './ImageBadge';

test('ImageBadge', () => {
  render(<ImageBadge>ImageBadge</ImageBadge>);
  expect(screen.getByTestId('imagebadge')).toBeInTheDocument();
  expect(screen.getByTestId('imagebadge')).toMatchSnapshot();
});
