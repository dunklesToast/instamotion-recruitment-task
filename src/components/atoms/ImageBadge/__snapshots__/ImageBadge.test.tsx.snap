// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBadge 1`] = `
.c1 {
  font-family: 'PT Sans',sans-serif;
  font-style: regular;
  font-weight: 400;
}

.c0 {
  background-color: #0B4256;
  border-bottom-left-radius: 4px;
  border-top-left-radius: 4px;
  color: white;
  display: inline-block;
  padding: 8px 32px 8px 8px;
}

<div
  class="c0"
  data-testid="imagebadge"
>
  <p
    class="c1"
  >
    ImageBadge
  </p>
</div>
`;
